torchtune/utils/collate.py

```diff
@@ -68,7 +68,7 @@ def padded_collate(
             (0, labels_seq_len - input_ids_seq_len),
             value=padding_idx,
         )
-    return {"tokens": input_ids, "labels": labels}
+    return {"tokens": input_ids.long(), "labels": labels.long()}
```

Comment:
and this deals with the empty line issue?

Reply:
yes, when the line is empty these tensors become float tensors. so even if the line is empty this will ensure that the model still trains.

I was thinking about whether we should add a warning if a sample is completely empty (== all padding). It will get entirely ignored in loss anyway but it may be useful for debugging.